filedetails.go

```diff
@@ -0,0 +1,24 @@
+package main
```

Comment:
It was nice to pull this out. Why don't you make splitting this out and relocating it into `internal/file` a separate PR? You could land that first if you want and rebase the rest of this.

Reply:
How do you suggest that I handle the call to warnError in Updateinfo?